library/core/src/main/java/com/google/android/exoplayer2/extractor/ts/Ac3Reader.java

```diff
@@ -188,18 +192,36 @@ private boolean skipToNextSync(ParsableByteArray pesBuffer) {
   private void parseHeader() {
     headerScratchBits.setPosition(0);
     Ac3Util.Ac3SyncFrameInfo frameInfo = Ac3Util.parseAc3SyncframeInfo(headerScratchBits);
-    if (format == null || frameInfo.channelCount != format.channelCount
-        || frameInfo.sampleRate != format.sampleRate
-        || frameInfo.mimeType != format.sampleMimeType) {
+    if ((format == null || frameInfo.sampleRate != format.sampleRate
+        || frameInfo.mimeType != format.sampleMimeType)
```

Comment:
I think we should avoid the case where two formats are output, one with the incorrect channel count, then a following one with the correct channel count. We should be waiting until we know what the format should be, and just outputting that.

Also, if "I,D" is an access unit, should the reader be outputting "I,D" in a single sample, rather than two separate samples as appears to be the case currently? Note this removes the need for setting sampleDurationUs to 0 below.

By the way, if you want to get the changes in the other two files submitted ahead of this discussion being concluded, feel free to send those as a separate pull request. I think they're ready to merge.

Thanks!